### ISSUE TITLE
The minimum mode of the key-spacing rule has a bug.

### ISSUE THREAD

**Issue by Alex1990** [eslint/eslint#5008]
**ESLint version:** the master branch (commit 0acbe57602b1297cc1e7a4b04edac8651ed378f9)

In production code, I use the 1.10.3.

**Lint result:**

```shell
cjc:~/www/node/eslint$ ./bin/eslint.js -c .eslintrc.json index.js 

/home/cjc/www/node/eslint/index.js
  1:1  error  Use the global form of 'use strict'  strict

✖ 1 problem (1 error, 0 warnings)
```

**`eslintrc.json`:**

```js
{
  "rules": {
    "key-spacing": [2, { "mode": "minimum", "beforeColon": false, "afterColon": true }]
  }
}
```

**`index.js`:**

```js
(function() {
    var obj = {
        foo : "Foo"
    };

    obj.foo = "Bar";
}());
```

There should not be an extra space before the colon. I mean an error should be output.

**Comment by nzakas** [comment]
@btmills ?

**Comment by btmills** [comment]
Based on https://github.com/eslint/eslint/blob/v1.10.3/tests/lib/rules/key-spacing.js#L164-L168, it looks like this might be by design. @BYK, is this behavior intentional?

**Comment by ilyavolodin** [comment]
@BYK could you provide information about this rule?

**Comment by BYK** [comment]
Not sure why I didn't get @btmills' direct mention. Anyways, this looks liek a bug. I was able to reproduce the behavior in the online demo. I'll see if I can get a patch up soon.
